### DISCUSSION TITLE
Solax X3 Hybrid G1/2 3 Phase Modbus

### DISCUSSION THREAD

**Discussion by Esplugas** [wills106/homeassistant-solax-modbus#1152] [General]
Hi, I'm trying to get this to work with a pair of Solax X3 Hybrid G1/2 3 Phase inverters without success.  They are both connected via the LAN dongles to my network.  The inverters are not compatible with the Solax Wifi dongle so it looks like Modbus connection is my only option.  Any advice would be much appreciated.

**Comment by wills106** [comment]
PocketLAN doesn't provide a Modbus connection. Your Inverters do have built in ethernet though.

**Reply by Esplugas** [reply]
The attached picture is the connection I have for each machine.  I tried the standard Solax integration in HA without success using the IP address for each machine as shown on my Unifi setup.  (192.168.1.90 and 118)  The confusion comes when I log into the Solaxcloud app (which works fine) as it says my IP connection is 192.168.1.8 for both machines but that doesn't work either as it belongs to another unrelated device on my network.  
![IMG_0416 copy](https://github.com/user-attachments/assets/35c360b1-7913-48c7-819b-2e5c87004ed9)

**Reply by wills106** [reply]
Have a look at https://github.com/wills106/homeassistant-solax-modbus/discussions/303

**Reply by Esplugas** [reply]
Thanks for that.  So technically I should be able to plug straight into the built in ethernet socket behind the front panel and use the HA Solax Integration?  I'll have a look and report back 👍

**Reply by wills106** [reply]
Yea and you can use that for the Cloud instead of the PocketLAN if you so wished.

**Reply by wills106** [reply]
You don't need a password to connect. MQTT is for the cloud.

You use the default port of 502 for Modbus. You are using this Integration and not the SolaX one built into Home Assistant aren't you?

![image](https://github.com/user-attachments/assets/d751790d-636b-40d7-83bc-b5417b4379cb)

**Reply by Esplugas** [reply]
It's working perfectly now, many thanks for your help.  One last question, as I have two inverters, master and slave, when I set up your integration do I select parallel mode (master / slave ) and connect to the master inverter IP or do I need to register two separate connections independently?
